src/lex/cursor.rs

```diff
@@ -88,7 +88,7 @@ impl<'a> Cursor<'a> {
         };
         self.col += ret.chars().count();
         self.index += remaining_end;
-        ret
+        ret.trim()
```

Comment:
That's good. For improvement, you can omit skipping whitespaces with the line 70-76 and 80-88. `ret.trim()` will remove them.

Reply:
Lines 80-88 should be kept, the error message will include `col`, for accuracy this check is necessary.